File: BackEnd/WeShare.Application/Dtos/Transaction/TransactionViewDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WeShare.Application.Dtos.Task;
using WeShare.Application.Dtos.TransactionSplit;
using WeShare.Application.Dtos.User;
using WeShare.Core.Entities;
using WeShare.Core.Enums;

namespace WeShare.Application.Dtos.Transaction
{
    public class TransactionViewDto
    {
        public int Id { get; set; }
        public int GroupId { get; set; }
        public UserViewDto Payer { get; set; }
        public decimal Amount { get; set; }
        public string? Description { get; set; }
        public TaskViewDto? Task { get; set; }
        public DateTime CreatedAt { get; set; }
        public SplitStrategyEnum SplitStrategy { get; set; }
        public IEnumerable<TransactionSplitViewDto> TransactionSplits { get; set; }
    }

}

File: FrontEnd/src/types/transaction/TransactionViewDto.ts
import type { SplitStrategyEnum } from "@/constants/SplitStrategyEnum"
import type { UserViewDto } from "../user/UserViewDto"
import type { TransactionSplitViewDto } from "./TransactionSplitViewDto"

export interface TransactionViewDto {
  id: number
  groupId: number
  payer: UserViewDto
  amount: number
  description?: string | null
  task?: number | null // Assuming task is represented by its ID, adjust if it's a different type
  splitStrategy: SplitStrategyEnum
  transactionSplits: TransactionSplitViewDto[]
  createdAt: Date
}
